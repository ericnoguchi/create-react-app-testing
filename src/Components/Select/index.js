import React from "react";

function Select(props) {
  return (
    <select
      data-testid={props["data-testid"]}
      className={props.className}
      value={props.value}
      onChange={props.onChange}
    >
      <option key="empty" value="">
        -- select --
      </option>
      {props.collection.map(item => (
        <option key={item[props.valueProp]} value={item[props.valueProp]}>
          {item[props.titleProp]}
        </option>
      ))}
    </select>
  );
}

export default Select;
