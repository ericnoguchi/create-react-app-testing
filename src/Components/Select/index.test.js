import { shallow } from "enzyme";
import React from "react";
import Select from ".";

describe("<Select /> rendering", () => {
  let props = {
    collection: [
      {
        valueProp: "value1",
        titleProp: "title1"
      },
      {
        valueProp: "value2",
        titleProp: "title2"
      }
    ],
    valueProp: "valueProp",
    titleProp: "titleProp",
    className: "testClassName",
    value: "testValue",
    onChange: jest.fn()
  };

  let Component;

  beforeEach(() => {
    props.onChange.mockClear();
    Component = <Select {...props} />;
  });

  // snapshot demo
  it("should render to match snapshot", () => {
    expect(shallow(Component)).toMatchSnapshot();
  });

  // simulate event and spy fn demo
  it("should should call props on change with value changes", () => {
    let wrap = shallow(Component);
    wrap.find("select").simulate("change", "changedValue");
    expect(props.onChange).toHaveBeenCalledWith("changedValue");
  });
});
