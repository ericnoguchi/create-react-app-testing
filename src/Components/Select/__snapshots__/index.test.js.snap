// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> rendering should render to match snapshot 1`] = `
<select
  className="testClassName"
  onChange={[MockFunction]}
  value="testValue"
>
  <option
    key="empty"
    value=""
  >
    -- select --
  </option>
  <option
    key="value1"
    value="value1"
  >
    title1
  </option>
  <option
    key="value2"
    value="value2"
  >
    title2
  </option>
</select>
`;
